app/models/form_version.rb

```diff
@@ -34,17 +35,20 @@ class FormVersion < ApplicationRecord
   after_initialize :generate_code, :generate_number
   before_create :ensure_unique_code, :ensure_unique_number
 
+  # Only one can be true per form.
+  validates :is_current, uniqueness: {allow_blank: true, scope: :form_id}
+  validates :is_oldest_accepted, uniqueness: {allow_blank: true, scope: :form_id}
```

Comment:
For future reference, Rails convention says you don't need `is_` in front of column names like this. Someday we can rename both.